internal/validators/namespace.go

```diff
@@ -107,6 +119,36 @@ func (v *Namespace) handle(req *nsRequest) admission.Response {
 	return allow("")
 }
 
+// illegalTreeLabel checks if tree labels are being created or modified
+// by any user or service account since only HNC service account is
+// allowed to do so
+func (v *Namespace) illegalTreeLabel(req *nsRequest) admission.Response {
+	msg := "Only HNC Service account can modify tree labels"
+	treeLabelID := "tree.hnc.x-k8s.io"
+
+	switch req.op {
+	case k8sadm.Update:
+		if !cmp.Equal(req.ns.Labels, req.oldns.Labels) {
```

Comment:
I'd skip this test. It's not actually any faster, and adds a new and unneeded dependency. You could just remove it completely and the code should still work identically.